app/models/task.py

```diff
@@ -1,5 +1,33 @@
 from app import db
 
-
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True, default=None)
+    
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'))
+    goal = db.relationship("Goal", back_populates="tasks")
+        
+    def to_dict(self):
+        return {
+            "id": self.task_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": (self.completed_at != None)
+        }
+        
+    def to_dict_with_goal_id(self):
+        task_dict = self.to_dict()
+        if self.goal_id:
+            task_dict["goal_id"] = self.goal_id
+        return task_dict
```

Comment:
Great work recycling the existing `to_dict()` function inside this helper function -- it makes the code so much DRY-er!